doc/devel/coding_style_guideline.rst

```diff
@@ -49,6 +49,182 @@ No alias should be used for `h5py`::
+--------------------
+Separate `cdef`, `cpdef`, and `ctypedef` statements from the following type by
+exactly one space. In turn, separate the type from the variable name by exactly
+one space. Declare only one `ctypedef` variable per line. You may `cdef` or
+`cpdef` multiple variables per line as long as these are simple declarations;
+note that multiple assignment, references, or pointers are not allowed on the
+same line. Grouping `cdef` statements is allowed. For example::
+
+  # Good
+  cdef int n
+  cdef char* s
+  cdef double Xf[3]
+  cdef double d[3]
+  cpdef int i, j, k
+  cdef Material mat = Material()
+  cdef:
+      double *ps = <double *> cnp.PyArray_DATA(seed)
+      cnp.npy_intp *pstr = <cnp.npy_intp *> qa.strides
+      cnp.npy_intp d, i, j, cnt
+      double direction[3]
+      double tmp, ftmp
+  cdef int true_enough(x):
+      return 1
+
+  # Bad
+  cdef  char *s
+  cdef char * s, * t, * u, * v
+  cdef double Xf[3], d[3]
+  cdef double x=42, y=x+1, z=x*y
+  cdef Material     mat   = Material()
+  cdef   int   falsified(x):
+      return 0
+
+Inside of a function, place all `cdef` statements at the top of the function
+body::
+
+  # Good
+  cdef int true_enough(x):
+      cdef int i = x
+      cdef int rtn
+      rtn = i + 42
+      return rtn
+
+  # Bad
+  cdef int falsified(x):
+      cdef int i = x, j = -42
+      j += i
+      cdef int rtn = j / j - 1
+      return rtn
+
+Using C libraries
+-----------------
+The `cimport`s should follow the same rules defined in PEP8 for `import`
+statements. If a module is both *imported* and *cimported*, the `cimport`
+should come before the `import`.
+
+An example of an imported C library::
+
+  from libc.stdlib cimport calloc, realloc, free
+
+Do not use `include` statements.
+
+Error return values
+-------------------
+When declaring an error return value with the `except` keyword, use one space on
+both sides of the `except`. If in a function definition, there should be no
+spaces between the error return value and the colon `:`. Avoid `except *`
+unless it is needed for functions returning `void`::
+
+  # Good
+  cdef void redwood() except *
+  cdef void c_extract(Feature self, Data2D datum, Data2D out) nogil except *:
+  cdef int sequoia(x) except +:
+      ...
+
+  # Bad
+  cdef char * spruce(x) except *:
+  cdef int fir(x)    except   +  :
+      ...
+
+Pointers and references
+-----------------------
+Pointers and references may be either zero or one space away from the type name.
+If followed by a variable name, they must be one space away from the variable
+name. Do not put any spaces between the reference operator `&` and the variable
+name::
+
+  # Good
+  cdef int& i
+  cdef char * s
+  i = &j
+
+  # Bad
+  cdef int &i
+  cdef char *s
+  i = & j
+
+Casting
+-------
+When casting a variable there must be no whitespace between the opening `<` and
+the type. There must one space between the closing `>` and the variable::
+
+  # Good
+  <float> i
+  <void *> s
```

Comment:
This may be subject to some discussion/is may be a matter of tastes. I guess the line [cythonutils.pyx:141](https://github.com/nipy/dipy/blob/c8d2f179dac618d9249c82a0053327f1edf1700b/dipy/segment/cythonutils.pyx#L141)
```
    memview._data = <char*> calloc(buffer_size, sizeof(float))
```
was may be not written following any specific guideline.

Reply:
I am ok, with this rule, we can update our code on a new PR. Any other thought?

Reply:
No. I'm OK with it. Yes, we should eventually converge to abide by these rules in our Cython code :blue_book: .